src/__tests__/a256gcmEncryption.test.ts

```diff
@@ -925,7 +926,197 @@ describe('ECDH-ES+A256KW (P-256) Auth, Key Wrapping Mode with A256GCM content en
       await expect(decryptJWE(jwe, recipients[0].decrypter)).rejects.toThrowError('Failed to decrypt')
       await expect(decryptJWE(jwe, recipients[0].decrypter)).rejects.toThrowError('Failed to decrypt')
     })
+    //// fail
+    it('Incompatible encrypters throw', async () => {
+      expect.assertions(1)
+      const enc1 = { enc: 'cool enc alg1' } as Encrypter
+      const enc2 = { enc: 'cool enc alg2' } as Encrypter
+      await expect(createJWE(cleartext, [enc1, enc2])).rejects.toThrowError('Incompatible encrypters passed')
+    })
+  })
+})
+
+// Adapted from:
+// https://github.com/decentralized-identity/veramo/blob/next/packages/did-comm/src/__tests__/encryption.test.ts#L492-L674
+// changes X25519 to P256
+describe('ECDH-ES+A256KW (P-256) Dir Auth, Key Wrapping Mode with A256GCM content encryption', () => {
+  describe('One recipient', () => {
+    let cleartext: Uint8Array, recipientKey: any, senderKey: any, decrypter: Decrypter, sharedSecret: Uint8Array
 
```

Comment:
Comment after DIF Hacakthon: It seems like unauthenticated encryption is the default for JWE with ECDH-ES+A256KW, so I should throw out any code that is '*'.Auth.'*' and just keep ''*'.Anon.'*'? 

i.e.: Keep AnonEncrypter, AnonDecrypter, AnonDirEncrypter and AnonDirDecrypter..

"JSON Object Signing and Encryption (JOSE) defines a number of
   encryption (JWE) [[RFC7516](https://datatracker.ietf.org/doc/html/rfc7516)] and digital signature (JWS) [[RFC7515](https://datatracker.ietf.org/doc/html/rfc7515)]
   algorithms.  When symmetric cryptography is used, JWE provides
   authenticated encryption that ensures both confidentiality and sender
   authentication.  However, for public key cryptography the existing
   JWE encryption algorithms provide only confidentiality and some level
   of ciphertext integrity.  When sender authentication is required,
   users must resort to nested signed-then-encrypted structures, which
   increases the overhead and size of resulting messages.  This document
   describes an alternative encryption algorithm called ECDH-1PU that
   provides public key authenticated encryption, allowing the benefits
   of authenticated encryption to be enjoyed for public key JWE as it
   currently is for symmetric cryptography."
   https://datatracker.ietf.org/doc/html/draft-madden-jose-ecdh-1pu-03

Reply:
Yes, only `-1PU` is auth, `-ES` is anon

Reply:
to-do: make sure https://github.com/decentralized-identity/did-jwt/blob/master/src/__tests__/jwe-vectors.ts is up to date to cover these changes.